### ISSUE TITLE
Renderer encountered an unexpected error: -2005270527

### ISSUE THREAD

**Issue by phealy** [microsoft/terminal#14082]
### Windows Terminal version

1.16.2642.0

### Windows build number

10.0.22000.978

### Steps to reproduce

1. Have a session open (this one is powershell running OpenSSH to a Linux VM)
2. Suspend the system
3. Wake the system

After resuming, I continuously get the following error: `Renderer encountered an unexpected error: -2005270527`. Dismissing it just causes it to come up again. I tried dismissing it about 20 times and it didn't clear.

### Expected Behavior

No errors.

### Actual Behavior

Repeated continuous popups of the following error:
![image](https://user-images.githubusercontent.com/3046014/192338807-ce253c28-55b8-4053-bfca-bb41360d49ac.png)

**Comment by lhecker** [comment]
Could you follow the steps in this comment to force enable the debug layer? https://github.com/microsoft/terminal/issues/13985#issuecomment-1246111446
But don't change the setting for the "Feature level limit".

Afterwards you should see more detailed error descriptions by simultaneously running [DebugView](https://docs.microsoft.com/en-us/sysinternals/downloads/debugview).

@DHowett Warning popups are unfortunately separate from the "error state" one.

**Comment by lhecker** [comment]
Note to self/team: The likely fix will be to set the device invalidation flag here:
https://github.com/microsoft/terminal/blob/e1e6e662f9d00d55770c2c947066d7de18ea286f/src/renderer/atlas/AtlasEngine.cpp#L516-L538

In normal operation there shouldn't ever be any thrown exceptions, so destroying devices eagerly shouldn't be as bad as it sounds.

**Comment by lycandroid** [comment]
I have this problem intermittently as well and am also running windows terminal preview 1.16.2642.0.

In my case I hibernate, as opposed to sleep, my machine.  I tend to find the error occurs when I resume a tmux session, but I just happened to get the problem just now when I went into vim from a local bash shell.  

What I find is that if I resize the main window after the error is shown, then it's not shown again for the rest of the session.

It might be related but I also sometimes find that, after resuming, my tmux session it doesn't seem to know the window size correctly, for example it puts the tmux status bar on the wrong row.  Again, resizing the main window fixes that problem.

**Comment by lhecker** [comment]
> ### Windows Terminal version
> 1.15.2713.0
> 
> ### Windows build number
> 10.0.22621.521
> 
> ### Other Software
> ssh'ed to AWS machine, used `vim` inside there
> 
> ### Steps to reproduce
> No idea. I had three terminals in one window arranged in a "pi" pattern:
> 
> ```
> --------
> content
> --------
> p1 | p2
> ```
> 
> I had made p1 bigger using ctrl-alt-up/right. I noticed when I hovered over the pane and exited, the bell tripped. everytime i entered the pane, it was fine, but when exiting the pane the bell would trigger
> 
> ### Expected Behavior
> No error and no bell firing
> 
> ### Actual Behavior
> When I went to vim, the renderer error triggered: Renderer encountered an unexpected error: -2005270527
> 
> Strangely when I made the p2 pane bigger again (see above for pane diagram) the error cleared and things worked fine. every key stroke triggered the error and exiting vim was "hard/impossible". Only after making the pane bigger, did the error go away and I could exit vim.

Source: https://github.com/microsoft/terminal/issues/14171

**Comment by lhecker** [comment]
@dovholuknf Could you confirm whether you have "AtlasEngine" enabled in your Rendering or advanced profile settings?

In case you do and if you encounter the issue again, could you please run [DebugView](https://docs.microsoft.com/en-us/sysinternals/downloads/debugview) and check if any errors are logged while the issue happens inside Windows Terminal?

**Comment by dovholuknf** [comment]
I had left my machine on, came back to it after it sleepings and I do not see "atlas" in my settings.json so I assume it's set to whatever 'the default' is? Here's what dbgview looked like at that time
 
![image](https://user-images.githubusercontent.com/46322585/194929546-0f00897f-a756-472d-9d9a-3e3c4dce97a3.png)

I also have a recording where the bell fired -- but NOT `error: -2005270527` which is potentially interesting. Another thing to note is that i have the "move focus when i mouse over" option set.

Here's a video if interested. You'll see I did not get the same error (yet) and during that time i had debug viewer open as well... https://youtu.be/AeW9xUTl4OY

**Comment by ilor** [comment]
I've gotten "Renderer encountered an unexpected error: -2005270527" twice now, it seems to be related to suspend and/or a semi-broken ssh session state. Both times I got it after I ran `vi` in a terminal session that was in a glitched state after a disconnect from an ssh session running vi.

**Comment by krzysdz** [comment]
I get this unexpected error when running `oh-my-posh debug` in maximized window (maximized before running the command) with *"AtlasEngine"* enabled.

DebugView shows the following message when the pop-up shows up:
```
[3676] DXGI ERROR: IDXGISwapChain::Present: The scroll rect #0 passed to Present is empty. ScrollRect = RECT{ left = 0, top = 0, right = 1888, bottom = 0 } [ MISCELLANEOUS ERROR #109: ]
```

Terminal version: ae7595b (Release, x64), can't reproduce in 1.16.10261.0
Windows build number: 19045.2846
oh-my-posh version: 15.4.0
<details>
<summary>oh-my-posh theme</summary>

```jsonc
{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "version": 2,
  "final_space": true,
  // "console_title_template": "{{ .Shell }} - {{ .UserName }}: {{ .PWD }}",
  "console_title_template": "{{ .UserName }}: {{ .PWD }}",
  "blocks": [
    {
      "type": "prompt",
      "alignment": "left",
      "segments": [
        {
          "type": "os",
          "style": "diamond",
          "foreground": "#fff",
          "background": "#003543",
          "leading_diamond": "\ue0b6",
          "template": " {{ if .WSL }}WSL at {{ end }}{{.Icon}}",
          "properties": {
            "windows": "\ue62a"
          }
        },
        {
          "type": "session",
          "style": "powerline",
          "powerline_symbol": "\ue0b0",
          "foreground": "#d2ff5e",
          "background": "#003543",
          "template": "{{ if .SSHSession }}\udb80\udf18 {{ end }}{{ .UserName }} ",
          "properties": {
            "display_host": false
          }
        },
        {
          "type": "path",
          "style": "powerline",
          "powerline_symbol": "\ue0b0",
          "foreground": "#003544",
          "background": "#0087D8",
          "template": " \ue5ff {{ path .Path .Location }} ",
          "properties": {
            "folder_separator_icon": "/",
            "style": "full"
          }
        },
        {
          "type": "git",
          "style": "powerline",
          "powerline_symbol": "\ue0b0",
          "foreground": "#193549",
          "background": "#d2ff5e",
          "background_templates": [
            "{{ if or (.Working.Changed) (.Staging.Changed) }}#ff9248{{ end }}",
            "{{ if and (gt .Ahead 0) (gt .Behind 0) }}#f26d50{{ end }}",
            "{{ if gt .Ahead 0 }}#89d1dc{{ end }}",
            "{{ if gt .Behind 0 }}#f17c37{{ end }}"
          ],
          "template": " {{ .UpstreamIcon }}{{ .HEAD }}{{ if .Staging.Changed }} \uf046 {{ .Staging.String }}{{ end }}{{ if and (.Working.Changed) (.Staging.Changed) }} |{{ end }}{{ if .Working.Changed }} \uf044 {{ .Working.String }}{{ end }}{{ if gt .StashCount 0 }} \udb80\udd93 {{ .StashCount }}{{ end }} ",
          "properties": {
            "fetch_stash_count": true,
            "fetch_status": true,
            "fetch_upstream_icon": true
          }
        },
        {
          "type": "exit",
          "style": "powerline",
          "powerline_symbol": "\ue0b0",
          "foreground": "#242424",
          "background": "#33DD2D",
          "background_templates": [
            "{{ if gt .Code 0 }}#f1184c{{ end }}"
          ],
          "template": " \udb81\udf8f {{ if gt .Code 0 }}\uf00d {{ .Code }}{{ else }}\uf42e{{ end }} ",
          "properties": {
            "always_enabled": true
          }
        }
      ],
      "vertical_offset": 1
    },
    {
      "type": "prompt",
      "alignment": "right",
      "segments": [
        {
          "type": "spotify",
          "style": "powerline",
          "powerline_symbol": "\ue0b2",
          "invert_powerline": true,
          "foreground": "#ffffff",
          "background": "#1BD760",
          "template": " \udb81\udcc7 {{ .Icon }}{{ if ne .Status \"stopped\" }}{{ .Artist }} - {{ .Track }}{{ end }} ",
          "properties": {
            "paused_icon": "\udb80\udfe4 ",
            "playing_icon": "\ue602 ",
            "stopped_icon": "\uf04d ",
            "track_separator": " - "
          }
        },
        {
          "type": "battery",
          "style": "powerline",
          "powerline_symbol": "\ue0b2",
          "invert_powerline": true,
          "foreground": "#242424",
          "background": "#f36943",
          "background_templates": [
            "{{ if eq \"Full\" .State.String }}#0476d0{{ end }}",
            "{{ if eq \"Charging\" .State.String }}#33DD2D{{ end }}",
            "{{ if eq \"Discharging\" .State.String }}#FFCD58{{ end }}"
          ],
          "template": " {{ if not .Error }}{{ .Icon }}{{ .Percentage }}{{ end }}{{ .Error }}\uf295 \udb80\udc84 "
        },
        {
          "type": "python",
          "style": "powerline",
          "powerline_symbol": "\ue0b2",
          "invert_powerline": true,
          "foreground": "#003544",
          "background": "#0087D8",
          "template": " \ue235 {{ if .Error }}{{ .Error }}{{ else }}{{ if .Venv }}{{ .Venv }} {{ end }}{{ .Full }}{{ end }} ",
          "properties": {
            "display_mode": "context",
            "fetch_version": true,
            "fetch_virtual_env": true
          }
        },
        {
          "type": "node",
          "style": "powerline",
          "powerline_symbol": "\ue0b2",
          "invert_powerline": true,
          "foreground": "#ffffff",
          "background": "#6CA35E",
          "template": " \ue718 {{ if .PackageManagerIcon }}{{ .PackageManagerIcon }} {{ end }}{{ .Full }} "
        },
        {
          "type": "rust",
          "style": "powerline",
          "powerline_symbol": "\ue0b2",
          "invert_powerline": true,
          "foreground": "#193549",
          "background": "#99908a",
          "template": " \ue7a8 {{ if .Error }}{{ .Error }}{{ else }}{{ .Full }}{{ end }} "
        },
        {
          "type": "executiontime",
          "style": "powerline",
          "powerline_symbol": "\ue0b2",
          "invert_powerline": true,
          "foreground": "#ffffff",
          "background": "#8800dd",
          "template": " <#fefefe>\udb81\udead</> {{ .FormattedMs }} ",
          "properties": {
            "threshold": 500,
            "style": "austin"
          }
        },
        {
          "type": "time",
          "style": "diamond",
          "invert_powerline": true,
          "foreground": "#fff",
          "background": "#003543",
          "trailing_diamond": "\ue0b4",
          "template": "<#fff> \udb82\udd54 </>{{ .CurrentDate | date .Format }} "
        }
      ]
    },
    {
      "type": "prompt",
      "alignment": "left",
      "segments": [
        {
          "type": "root",
          "style": "plain",
          "foreground": "#FFD700",
          "template": " \uf0e7 "
        },
        {
          "type": "text",
          "style": "plain",
          "foreground": "#f1184c",
          "template": "\ud83d\ude80"
        }
      ],
      "newline": true
    }
  ]
  // "transient_prompt": {
  //   "background": "transparent",
  //   "foreground": "#FFD700",
  //   "template": "{{if .Root}}\u26a1 {{end}}🚀 "
  // }
}

```

Theme uses symbols from Nerd Fonts v3 (used font was CaskaydiaCove Nerd Font), but the bug can be reproduced even with the default font.

</details>

![Maximized Terminal Dev with transparency and a "Renderer encountered an unexpected error: -2005270527" pop-up, behind it a DebugView and dxcpl windows are open, VS 2022 Preview open on the second screen](https://user-images.githubusercontent.com/12915102/236087444-cb31e233-47b9-4d4d-b330-a050ab64592e.png)

**Comment by lhecker** [comment]
I fixed that issue here: #15262
It’s a somewhat annoying bug, so I’ll try to get it merged soon. I‘m super happy to see you try out the main branch btw! I hope there aren’t any serious issues apart from that one. 😊

**Comment by jimf5** [comment]
Got this issue "Renderer encountered an unexpected error: -2005270527" on the latest Terminal Preview [v1.18.1421.0]
Repro: start nvim in WSL ubuntu

**Comment by lhecker** [comment]
Those repro steps don't reproduce the issue for me. Could you please provide more detailed steps? nvim config, font settings, display scale, window size, etc., anything you think is relevant to reproduce the issue on another PC basically.

Just to be sure though, if you're using some custom font, could you try and see if the issue reproduces with Cascadia Mono or Cascadia Code? 🤔

**Comment by dovholuknf** [comment]
For me, it just seems to happen "randomly" and without any real indication of what triggers it. It's happened when I used cmd.exe from terminal, it's happened from bash, it's happened from ubuntu 20 from ubuntu 22... i can't figure out how to reproduce it _at all_...

I suspect some character/escape sequence is rendered to some pane somewhere and triggers it. It does seem to only happen when i have "split" the terminal though. I don't recall it happening when the terminal wasn't split. 

It often occurs after waking from a lock state/sleep state. I'm on the latest preview, it's much less frequent, but it still happens. :(

**Comment by lhecker** [comment]
If the issue happens often enough, one option is to enable the debug layer and listen for debug messages. This however only works if you have the "Graphics Tools" feature installed (Settings app > "Apps" > "Optional Features"). If you want to try that, here's how to do it:

1. First enable the debug layer:
   * On modern Windows you can run:
     ```pwsh
     d3dconfig apps --add WindowsTerminal.exe
     d3dconfig debug-layer debug-layer-mode=force-on
     ```
   * On older Windows:
     Run `dxcpl` from the start menu and click this:
     ![image](https://github.com/microsoft/terminal/assets/2256941/36b9a1d7-fc4c-4972-b77c-d7c33736924e)
2. Listen for debug output via [DbgView](https://learn.microsoft.com/en-us/sysinternals/downloads/debugview).
3. Disable the debug layer
   * New:
     ```pwsh
     d3dconfig --reset
     ```
   * Old:
     ![image](https://github.com/microsoft/terminal/assets/2256941/f93fc613-e4e6-49b9-8751-40012e909f2e)

**Comment by jimf5** [comment]
> If the issue happens often enough, one option is to enable the debug layer and listen for debug messages. This however only works if you have the "Graphics Tools" feature installed (Settings app > "Apps" > "Optional Features"). If you want to try that, here's how to do it:
> 
>     1. First enable the debug layer:
>        
>        * On modern Windows you can run:
>          ```powershell
>          d3dconfig apps --add WindowsTerminal.exe
>          d3dconfig debug-layer debug-layer-mode=force-on
>          ```
>        * On older Windows:
>          Run `dxcpl` from the start menu and click this:
>          ![image](https://user-images.githubusercontent.com/2256941/240584691-36b9a1d7-fc4c-4972-b77c-d7c33736924e.png)
> 
>     2. Listen for debug output via [DbgView](https://learn.microsoft.com/en-us/sysinternals/downloads/debugview).
> 
>     3. Disable the debug layer
>        
>        * New:
>          ```powershell
>          d3dconfig --reset
>          ```
>        * Old:
>          ![image](https://user-images.githubusercontent.com/2256941/240585347-f93fc613-e4e6-49b9-8751-40012e909f2e.png)

I've got this upon error:
00000001	0.00000000	[18696] DXGI ERROR: IDXGISwapChain::Present: Partial Presentation (using scroll or dirty rects) is not valid until first submitting a regular Present without scroll or dirty rects. Otherwise, the preserved back-buffer data would be uninitialized. [ MISCELLANEOUS ERROR #105: ]

**Comment by chrisl8** [comment]
As of today, Windows Terminal has become entirely unusable due to this error.

If I Win+Arrow key to move the terminal window, it starts spamming this error repeatedly and the only solution is to close the Terminal.

That said, the terminal doesn't really work well even if I don't Win+arrow it. The cursor is broken and I cannot type.

Windows Terminal Preview
Version: 1.18.1421.0

I will try moving back to the release version. I've been on the Preview for so many years I kind of forgot it was Beta.

**Comment by zadjii-msft** [comment]
You can also just try:

```json
    "profiles":
    {
        "defaults":
        {
            "useAtlasEngine": false
            
```

in the settings. That should switch back to the old renderer

**Comment by lhecker** [comment]
@chrisl8 That issue with the cursor is fixed in #15403. I'm sorry for causing the issue - We only noticed it mere hours before publishing 1.18. It's because of your particular font, which most likely behaves similarly to MesloLGM NF. I failed to test that particular aspect (= "What if the glyphs in the first row are outside of the bounds of the window"?).
You don't describe the cursor issue in detail, but I found that I can replicate it with the Direct2D based text renderer in AtlasEngine. That one is only used when a device does not have a GPU. Are you perhaps using a VM or remote desktop? I've fixed that in #15427.

@jimf5 That's super helpful and narrows it down _a lot_. Although even that I now know which couple dozen lines should be responsible for this, I can't quite figure out _why_ it would happen. You don't happen to use MesloLGM NF or a similar nerd font, etc., either, right?

I've built a Windows Terminal Preview version that integrates the previous 3 AtlasEngine bug fixes, 2 of which might be relevant here: #15403 and #15425. (I did the cursor fix after starting the hour-long build and signing process.)
I don't want to waste anyone's time, but if you don't mind, I'd be super happy if you could check it out and see if it fixes your issues. You can download it here: https://blob.hecker.io/sharex/Microsoft.WindowsTerminalPreview_1.18.1443.0_8wekyb3d8bbwe.msixbundle
I'm using my own personal domain because I don't want to expose my internal work addresses. However the file is properly cryptographically signed by Microsoft.

FYI you can enable/disable AtlasEngine here at any time as well:
![image](https://github.com/microsoft/terminal/assets/2256941/c5ee4574-301f-486c-af26-cb9b0316674a)

**Comment by dovholuknf** [comment]
I've changed to useAtlasEngine false and today - I dunno if graphics drivers updted or what but I no longer can split tabs _at all_.

In DbgView I see:
```
[3680] WrappedMessage::OpenProperty
[3680]  [
[3680] store_provider\wrapped_message.cc
[3680]  @ 
[3680] 321
[3680] ]: 
[3680] Unknown error 0x8004010f
```

The moment I cross the split pane, it starts to crash and then just disappears without a trace
![splitting-fail](https://github.com/microsoft/terminal/assets/46322585/ae40a69c-23bf-4c6a-a68e-e2b951ebb570)

**Comment by dovholuknf** [comment]
I see Terminal Preview updated today in my Installed Apps list too... Can't be a coincidence? 
![image](https://github.com/microsoft/terminal/assets/46322585/91788a10-2207-4472-b45e-02affb7c8819)

**Comment by lhecker** [comment]
@dovholuknf Does the issue happen if you disable AtlasEngine, as described in the screenshot right above your comment? Or does the very-preview build I posted there fix your issue?
I'm not 100% sure but I feel like the `0x8004010f` error is unrelated to Windows Terminal. A crash dump would probably be helpful and the way to get one is described here among others: https://github.com/microsoft/terminal/wiki/Toubleshooting-Tips#capture-automatically
Otherwise, if you have diagnostic data enabled in Windows, we'll hopefully receive stack traces of the crash of it soon.

**Comment by dovholuknf** [comment]
Both. It's _really_ bad... unusable

![splitting-fail2](https://github.com/microsoft/terminal/assets/46322585/b1e4f90b-60fc-42e6-9881-7c1416f848ba)

**Comment by lhecker** [comment]
I should've mentioned that you need to create a new tab or restart the app after making that change. (I'll make myself a reminder to add a subtitle note to that setting, pointing that out.)

**Comment by dovholuknf** [comment]
I just tried your "totally not sketchy" build (i verified it was signed by MS :) ) and it crashes...

![splitting-fail3](https://github.com/microsoft/terminal/assets/46322585/5f7d7ddf-b4f6-4a71-a01a-699541f4d41d)

**Comment by dovholuknf** [comment]
I have a crash dump, if i put share it publically can you grab it "now" so I can 'unshare' it?

**Comment by lhecker** [comment]
Ah damn, I just realized what your issue most likely is. I even reviewed the PR that fixes the issue today. 😅 To temporarily fix it, please disable `focusFollowMouse` in your settings file. It's tracked by #15412. I'm sorry for wasting your time, but I'm incredibly thankful for walking me through your issue! 🙂

**Comment by dovholuknf** [comment]
that makes it 'not crash' indeed. i like the focusFollowMouse feature tho! :(

**Comment by jimf5** [comment]
> 

@lhecker, I'm using "Fira Code Nerd Font" and have "Atlas engine" turned on.  Turning off "Atlas" solves the issue.

**Comment by lhecker** [comment]
Yep that's for sure. The way the old text renderer uses `IDXGISwapChain::Present` is a lot more primitive. But does using "Atlas" with the custom preview build I linked above solve the issue? It's okay if you don't want to use such a custom version of course. 🙂 We'll likely release an update sometime soon. We do intend to make the new text renderer the default with 1.19 and any help we can get to test it would be extremely appreciated!

**Comment by darovic** [comment]
> You can download it here: https://blob.hecker.io/sharex/Microsoft.WindowsTerminalPreview_1.18.1443.0_8wekyb3d8bbwe.msixbundle

I can confirm that this build fixed the above issue for me when using the nerd font patched version of Cascadia Code. Thank you!

**Comment by chrisl8** [comment]
@lhecker This is on a desktop with an Nvidia 3080, so definitely have a GPU. It is late here, but I will definitely try out these updates in the morning and provide feedback.

**Comment by lhecker** [comment]
FYI: If you use "Caskaydia Cove" from the Nerd Fonts website, I found that it had a lot (really, ***a lot***) of subtle issues at least until a few months ago, whereas [Delugia](https://github.com/adam7/delugia-code) is almost flawless and extremely close to the original Cascadia Code. [Delugia](https://github.com/adam7/delugia-code) is sort of what Caskaydia Cove is based on, and I can only recommend trying it out.

---

@chrisl8 Ah I think I misunderstood you. The way you describe this issue:
> However, I still get a "filled box" cursor in 18, regardless of what I set the cursor to, while I get the requested Bar cursor in 17.

made it sound like #15427. But what you wrote afterwards:
> It seems I have a lot of profiles, some show up in the JSON but not the GUI and fiddling with the "cursorShape" entry in the JSON fixes this, even though fiddling with it in GUI does not.

makes it instead sound like you've been victim of our somewhat confusing "settings layering" where the per-profile settings "override" the settings in the "Defaults" profile and you need to reset the per-profile ones for the "Defaults" to take effect. So I guess it's not really a rendering issue after all!

Otherwise, you can always confirm if you use the full-featured, performant Direct3D renderer, or the slower Direct2D renderer by running this in a new PowerShell tab:
```pwsh
"$([char]0x1b)]11;rgb:7f/7f/7f$([char]0x1b)\"
```
It doesn't matter if it's the old "powershell.exe" or the new "pwsh.exe". If the cursor is visible afterwards, you're using Direct3D and if it's invisible Direct2D.

**Comment by chrisl8** [comment]
@lhecker  Your posted version Version: 1.18.1443.0 seems to fix everything.

To be clear I was facing at least 3 distinct issues:

1. If I snapped the Window to any side of the screen with Win+Arrow it would immediately crash with the -2005270527 error as titled in this issue.
2. When initially started the terminal would only show bars or blocks instead of text. Texted typed would also be the same, and output to the terminal was blank, and then it would magically resolve itself after some time, or when I pressed Print Screen or opened the snipping tool (probably switching to any other window would fix it?). So I couldn't screen shot it.
3. The "Filled Block" cursor was showing up when I thought I had selected "Bar", which you've described was a config issue.

I think I was getting confused by 2 and 3 together, because I thought the series of blocks/bars was my cursor, but it was just what the terminal was spitting out instead of the normal prompt of my text.

In PowerShell if I run the given `"$([char]0x1b)]11;rgb:7f/7f/7f$([char]0x1b)\"` command the screen turns grey with light grey text, but the cursor is visibly flashing.

Everything seem entirely fixed now with 1.18.1443.0 while using the AtlasEngine.

**Comment by mikeggh** [comment]
> > You can download it here: https://blob.hecker.io/sharex/Microsoft.WindowsTerminalPreview_1.18.1443.0_8wekyb3d8bbwe.msixbundle

I had the same exact problem today, and this update works great!   Thanks abunch!

**Comment by darovic** [comment]
> FYI: If you use "Caskaydia Cove" from the Nerd Fonts website, I found that it had a lot (really, _**a lot**_) of subtle issues at least until a few months ago, whereas [Delugia](https://github.com/adam7/delugia-code) is almost flawless and extremely close to the original Cascadia Code. [Delugia](https://github.com/adam7/delugia-code) is sort of what Caskaydia Cove is based on, and I can only recommend trying it out.

Good to know, thank you! I have recently learnt that patched fonts generally aren't a great idea, but unless supporting multiple/specific fonts was implemented for a specified set of characters we're probably stuck with them for a while. 

Thought I'd mention, still going strong several days later without issues, so the build you linked is definitely doing something right.

**Comment by CJCombrink** [comment]
> > > You can download it here: https://blob.hecker.io/sharex/Microsoft.WindowsTerminalPreview_1.18.1443.0_8wekyb3d8bbwe.msixbundle
> 
> I had the same exact problem today, and this update works great! Thanks abunch!

Same issue with "Cascadia Mono" font and the preview update solves the issue

**Comment by lhecker** [comment]
If anyone is still experiencing this issue in the latest Terminal versions, please let me know! I'll close this issue as resolved in a week otherwise. 🙂

**Comment by dkull** [comment]
Still happening in 1.18.3181.0
After every resize, often after regaining focus when changing tabs. does not crash, just a dismissable notification. but happens constantly.
Atlas engine is on. I'm using ssh sessions and tmux, but it also errors in tabs without ssh or tmux.

**Comment by lhecker** [comment]
@dkull Could you open a new issue in that case? First, before anything else, please download [DebugView](https://learn.microsoft.com/en-us/sysinternals/downloads/debugview) and run it as Admin.

Inside a cmd or pwsh prompt run this:
```pwsh
d3dconfig apps --add WindowsTerminal.exe
d3dconfig debug-layer debug-layer-mode=force-on
```

Prepare to reproduce your issue and press Ctrl+X in DebugView. Then cause the issue and check if there's any debug logging output. It should look like this:
![image](https://github.com/microsoft/terminal/assets/2256941/412c1c51-ed09-4461-a8b0-e40d1c56ad50)

If you see such a message, please copy it (there's no context menu - you need to press Ctrl+C) and attach it to the issue.

To restore the D3D config now run this:
```pwsh
d3dconfig --reset
```

If there's no such message, please list any steps someone would need to take to reproduce the issue. It would be great if it also happens with cmd.exe, since that's always easiest to reproduce issues with. Your GPU information would be helpful too. You can get it by running `Get-CimInstance win32_VideoController` in PowerShell (any version).

**Comment by zou000** [comment]
Hope this would help someone in future...
`Renderer encountered an unexpected error: -2005270527` happened for me ~80% of time after I reboot the machine. When this happens, one of my old games `Grim Fandango` would fail to start as well. But if I start another old game `Half Life` and immediately quit, both `Windows Terminal` and `Grim Fandango` would work again.
After some digging and tracing, I believe this is caused by GPU compiler caching. Now whenever this happens, I just need to delete `AppData\Local\AMD\DxcCache` and everything would be back to normal again.

**Comment by zhaoshenhao** [comment]
I got same problem with Version: 1.19.10573.0 after a few updates, including a bios firmware update. I found a work-around for my case:
In Rendering setting,
1. turn off Use the new text renderer ("AtlasEngine").
2. turn on Redraw entire screen when display updates.
Hope that works for you as well.
